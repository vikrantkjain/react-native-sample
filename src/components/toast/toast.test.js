import Toast from './index';
import {fireEvent, render, screen} from '@testing-library/react-native';
import {Pressable} from 'react-native';
import renderer from 'react-test-renderer';
import {act} from 'react-test-renderer';
describe('Test Toast component', () => {
  const ref = {
    current: {
      show: jest.fn(),
    },
  };
  const tree = render(
    <Pressable testID="TEST-1" onPress={() => ref.current.show()}>
      <Toast ref={ref} />
    </Pressable>,
  );
  it('Render successfully', () => {
    expect(tree.toJSON()).toMatchSnapshot();
  });

  act(() => {
    fireEvent.press(screen.getByTestId('TEST-1'));
    expect(ref?.current?.show());
  });
  // it('call ref ', () => {
  //   fireEvent.press(screen.getByTestId('TEST-1'));
  //   expect(ref?.current?.show());
  // });
});
