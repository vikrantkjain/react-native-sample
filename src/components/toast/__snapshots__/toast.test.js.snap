// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Toast component Render successfully 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  testID="TEST-1"
>
  <View
    pointerEvents="auto"
    style={
      [
        {
          "alignItems": "center",
          "elevation": 999,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "zIndex": 10000,
        },
        {
          "top": 1214,
        },
      ]
    }
  >
    <View
      collapsable={false}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "black",
          "borderRadius": 5,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "opacity": 1,
          "padding": 10,
          "paddingHorizontal": 20,
        }
      }
    >
      <Text
        style={
          {
            "color": "white",
          }
        }
      />
    </View>
  </View>
</View>
`;
